update-systemd-network.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+#
+# Parses DHCP options from openvpn to update resolv.conf
+# To use set as 'up' and 'down' script in your openvpn *.conf:
+# up /etc/openvpn/update-resolv-conf
+# down /etc/openvpn/update-resolv-conf
+#
+# Used snippets of resolvconf script by Thomas Hood <jdthood@yahoo.co.uk>
+# and Chris Hanson
+# Licensed under the GNU GPL.  See /usr/share/common-licenses/GPL.
+# 02/2016 wgh@torlan.ru modified script to include systemd-resolved support
+# 07/2013 colin@daedrum.net Fixed intet name
+# 05/2006 chlauber@bnc.ch
+
+SYSTEMD_PREFIX=/run/systemd
+
+if [ ! -d "$SYSTEMD_PREFIX" ]; then
+    echo "$SYSTEMD_PREFIX doesn't exist" >&2
+    exit 1
+fi
+
+mkdir -p "${SYSTEMD_PREFIX}/network"
+
+IFNAME=$1
+
+NETWORK_FILE="${SYSTEMD_PREFIX}/network/openvpn_${IFNAME}.network"
+
+case $script_type in
+up)
+  for optionname in ${!foreign_option_*} ; do
+    option="${!optionname}"
+    echo $option >&2
+    part1=$(echo "$option" | cut -d " " -f 1)
+    if [ "$part1" == "dhcp-option" ] ; then
+      part2=$(echo "$option" | cut -d " " -f 2)
+      part3=$(echo "$option" | cut -d " " -f 3)
+      if [ "$part2" == "DNS" ] ; then
+        IF_DNS_NAMESERVERS="$IF_DNS_NAMESERVERS $part3"
+      fi
+      if [[ "$part2" == "DOMAIN" || "$part2" == "DOMAIN-SEARCH" ]] ; then
+        IF_DNS_SEARCH="$IF_DNS_SEARCH $part3"
+      fi
+    fi
+  done
+
+  echo "IF_DNS_NAMESERVERS=$IF_DNS_NAMESERVERS" >&2
+  echo "IF_DNS_SEARCH=$IF_DNS_SEARCH" >&2
+
```

Comment:
Same here, does this echo does have any propose besides debugging? In that case. I think ``logger`` and send to system logging would be a better option.